syntax = "proto3";

package backend;

import "google/protobuf/empty.proto";

service Backend {
  rpc UpdateStudent(Student) returns (Empty) {}
  rpc RetrieveStudent(Student) returns (Student) {}

  rpc UpdateInstitution(Institution) returns (Empty) {}
  rpc RetrieveInstitution(Institution) returns (Institution) {}

  rpc UpdateProfileImage(ProfileImage) returns (Empty) {}
  rpc RetrieveProfileImage(Student) returns (ProfileImage) {}
}

message Student {
  string studentId = 1;
  string value = 2;
}

message Institution {
  string institutionId = 1;
  repeated Student students = 2;
  string encryptedInstitutionKey = 3;
  string institutionKeyIv = 4;
  string institutionName = 5;
  string passwordKeyNonce = 6;
  string verificationKey = 7;
}

message ProfileImage {
  string studentId = 1;
  // image stuff = 2;
}

message Empty {}
